clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/CTInAppBaseFragment.java

```diff
@@ -54,7 +56,11 @@ public void onAttach(Context context) {
             config = bundle.getParcelable(Constants.KEY_CONFIG);
             currentOrientation = getResources().getConfiguration().orientation;
             generateListener();
-            didClickForHardPermissionListener = (DidClickForHardPermissionListener) getActivity();
+            /*Initialize the below listener only when in app has InAppNotification activity as their host activity
+            when requesting permission for notification.*/
+            if (context instanceof InAppNotificationActivity) {
```

Comment:
@william-ct Please use either `getActivity()` or `context` for if block. Why are you using mixed? I know `context` in `onAttach()` is same as `getActivity()` but why to take risk?

Reply:
@william-ct Also you can just say `context  instanceof DidClickForHardPermissionListener`